### PR TITLE
Fixed CredentialsWrapper not passing authHandler to credentials fetcher

### PR DESCRIPTION
**Pull request by norberttech**
Whole description of this issue is available [here](https://github.com/googleapis/gax-php/issues/548)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -20,7 +20,9 @@
     "require-dev": {
         "phpunit/phpunit": "^9.6",
         "squizlabs/php_codesniffer": "3.*",
-        "phpspec/prophecy-phpunit": "^2.1"
+        "phpspec/prophecy-phpunit": "^2.1",
+        "ext-protobuf": "*",
+        "ext-grpc": "*"
```
Comment (norberttech) [inline composer.json:25]: without those two it seems to be impossible to run test suite, also it's anyway required by github workflows

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -40,8 +40,6 @@ jobs:
                 max_attempts: 3
                 command: composer update ${{ matrix.composerflags }}
             - name: Run script
-              env:
-                GOOGLE_APPLICATION_CREDENTIALS: ${{github.workspace}}/tests/Tests/Unit/testdata/json-key-file.json
```
Comment (norberttech) [inline .github/workflows/tests.yml:44]: same, without checking workflow it's pretty much impossible to figure out what's wrong so I just moved it to phpunit.xml.dist

**Comment by norberttech** [comment]
> Thanks for your pull request! It looks like this may be your first contribution to a Google open source project. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> View this [failed invocation](https://github.com/googleapis/gax-php/pull/555/checks?check_run_id=24096567273) of the CLA check for more information.
> 
> For the most up to date status, view the checks section at the bottom of the pull request.

lol that's a new level of corporation world bullshit but yeah, here you go: 

image

**Comment by norberttech** [comment]
> I'm not able to understand why it's necessary to add the grpc and protobuf extensions mandatory in the require-dev. I tried to fix this and PR #557 works without any of the configuration changes.

It's not necessary, but without it running locally `vendor/bin/phpunit` is failing, so it should be added to the composer.json required-dev section or the tests that are relaying on those two extensions should be marked as skipped when extensions are not present

File: composer.json (hunk 1 above)
Comment (bshaffer) [reply composer.json:25]: These test changes are really useful! Can we get them in a separate PR?

Comment (norberttech) [reply composer.json:25]: do you prefer to keep extensions as required-dev or maybe detect if the extension is available and mark tests that rely on it as skipped?

Comment (bshaffer) [reply composer.json:25]: I think the latter provided a better user experience

Comment (norberttech) [reply composer.json:25]: I think so too, will try to propose something later

**Pull request closed without merging**